libsweep/include/sweep/sweep.h

```diff
@@ -10,12 +10,28 @@
 extern "C" {
 #endif
 
+#if defined _WIN32 || defined __CYGWIN__ || defined __MINGW32__
+// If we are building a dll, set SWEEP_API to export symbols
+#ifdef BUILDING_DLL
```

Comment:
What is the use-case for making it conditional on `BUILDING_DLL`? (Also please prefix with `SWEEP_`)

Check slide 35 here for what we basically want to mimic
http://www.slideshare.net/StefanusDuToit/cpp-con-2014-hourglass-interfaces-for-c-apis